### PR TITLE
WOOC-1353 : Oney blocks

### PR DESCRIPTION
**Pull request by GhDj**

### CODE REVIEW

File: src/Gateway/Blocks/PayplugOney4x.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	 * Payment method name/id/slug.
+	 *
+	 * @var string
+	 */
+	protected $name = "oney_x4_with_fees";
+
+	protected $icon = 'x4_with_fees.svg';
+
+
+	public function oney_enabled() {
```
Comment (PPmmesquita) [inline src/Gateway/Blocks/PayplugOney4x.php:17]: this should be name `get_payment_method_data` shouldn't it?

File: src/Gateway/Blocks/PayplugOney4xWithoutFees.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	/**
+	 * Payment method name/id/slug.
+	 *
+	 * @var string
+	 */
+	protected $name = "oney_x4_without_fees";
+
+	protected $icon = 'x4_without_fees_';
+
+	public function oney_enabled() {
```
Comment (PPmmesquita) [inline src/Gateway/Blocks/PayplugOney4xWithoutFees.php:18]: this should be name `get_payment_method_data` shouldn't it?

File: src/Gateway/Blocks/PayplugOney.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	protected $name = "oney";
+
+	protected $icon = '';
+
+	protected $cart;
+
+	protected $total_price;
+
+
+	public function oney_enabled() {
```
Comment (PPmmesquita) [inline src/Gateway/Blocks/PayplugOney.php:21]: this should be name `get_payment_method_data` shouldn't it?

File: resources/js/frontend/wc-payplug-oney_x3_with_fees-blocks.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+if (typeof oney_response.x3_with_fees !== 'undefined') {
+	var down_payment_amount = parseFloat(settings?.oney_response['x3_with_fees']['down_payment_amount']) ;
+	var total_price_oney = down_payment_amount;
+	settings?.oney_response['x3_with_fees']['installments'].forEach((amount) => {
+
+		total_price_oney += parseFloat(amount['amount']);
+
+	});
+
+	Content = (e) => {
```
Comment (PPmmesquita) [inline resources/js/frontend/wc-payplug-oney_x3_with_fees-blocks.js:29]: Couldn't we have a generic OneySimulation Component with the values as parameter which would be used on the render?

File: src/Gateway/Blocks/PayplugOney.php (hunk 3 above)
Comment (GhDj) [reply src/Gateway/Blocks/PayplugOney.php:21]: PayplugOney is oney a parent class, the name property is being deprecated by the children classes

File: src/Gateway/Blocks/PayplugOney3xWithoutFees.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	 * Payment method name/id/slug.
+	 *
+	 * @var string
+	 */
+	protected $name = "oney_x3_without_fees";
+
+	protected $icon = 'x3_without_fees_';
+
+
+	public function oney_enabled() {
```
[the remark this answers is not in the document]
Comment (GhDj) [reply src/Gateway/Blocks/PayplugOney3xWithoutFees.php:19]: it is being called on the as `return $this->oney_enabled()` on the `get_payment_method_data` in the parent class

File: src/Gateway/Blocks/PayplugOney4x.php (hunk 1 above)
Comment (GhDj) [reply src/Gateway/Blocks/PayplugOney4x.php:17]: it is being called on the as `return $this->oney_enabled()` on the `get_payment_method_data`

File: src/Gateway/Blocks/PayplugOney4xWithoutFees.php (hunk 2 above)
Comment (GhDj) [reply src/Gateway/Blocks/PayplugOney4xWithoutFees.php:18]: it is being called on the as `return $this->oney_enabled()` on the `get_payment_method_data` in the parent class

**Review by PPmmesquita — APPROVED** [review]

**Comment by PPmmesquita** [comment]
I've re-do all the point mention above on the conversation, that's the reason I've closed the conversarion

**Pull request merged**